cli/cmd/endpoints.go

```diff
@@ -164,22 +161,31 @@ func requestEndpointsFromAPI(client destinationPb.DestinationClient, token strin
 					return
 				}
 
-				event, err := rsp.Recv()
-				if err != nil {
-					if grpcError, ok := status.FromError(err); ok {
-						err = errors.New(grpcError.Message())
+				// Endpoint state may be sent in multiple messages so it's not
+				// sufficient to read only the first message. Instead, we
+				// continuously read from the stream. This goroutine will never
+				// terminate if there are no errors, but this is okay for a
+				// short lived CLI command.
+				for {
+					event, err := rsp.Recv()
+					if errors.Is(err, io.EOF) {
+						return
+					} else if err != nil {
+						if grpcError, ok := status.FromError(err); ok {
+							err = errors.New(grpcError.Message())
+						}
+						errs <- err
+						return
 					}
-					errs <- err
-					return
+					events <- event
 				}
-				events <- event
 			}
 		}(authority)
 	}
-	// Block till all goroutines above are done
-	wg.Wait()
+	// Wait an amount of time for some endpoint responses to be received.
+	time.Sleep(5 * time.Second)
```

Comment:
My gut feeling is that this work group is fine here. I think depending on time.Sleep has one problem that you might encounter - if you hit an error while consuming from the stream, you will fail to report it right away but would still need to wait 5 seconds. Here is an alternative approach. 

1. Keep the workgroup
2. Pass the worklgroup to the streaming goroutine along with a context with a timeout of 5 seconds.
3. Keep reading in this routine until you hit an error or the context elapses (need to do a select over the stream chan and the context expiration chan)
4. Whatever happens (error or context elapsed), before exiting the goroutine call Done on the workgroup.
5. Keep the rest of the logic the same, wait for the workgroup, then read the errors and events.

Reply:
Unfortunately, I don't think that will work because the stream is not a chan; `rsp.Recv()` is a synchronous blocking call.

Instead, I've replaced the Sleep with a Timer so that we can select over the timeout, the messages, and the errors.  This lets us exit early if there is an error instead of needing to wait the entire timeout.